examples/MvcWeb/TheraLang/Services/ProjectTypeService.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using MvcWeb.Services;
+using MvcWeb.TheraLang.Entities;
+using MvcWeb.TheraLang.UnitOfWork;
+
+namespace MvcWeb.TheraLang.Services
+{
+    public class ProjectTypeService : IProjectTypeService
+    {
+        private readonly IUnitOfWork _uow;
+        public ProjectTypeService(IUnitOfWork unitOfWork)
+        {
+            _uow = unitOfWork;
+        }
+
+        public async Task AddProjectType(ProjectType projectType)
+        {
+            ProjectType Projecttype = new ProjectType { TypeName = projectType.TypeName };
+            if (projectType == null)
+            {
+                throw new NullReferenceException($"{nameof(ProjectType)} can`t be null");
+            }
+            try
+            {
+                await _uow.Repository<ProjectType>().Add(projectType);
+                await _uow.SaveChangesAsync();
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Error when trying to add new {nameof(ProjectType)}");
```

Comment:
add an exception to inner:
```suggestion
                throw new Exception($"Error when trying to add new {nameof(ProjectType)}", e);
```